### PR TITLE
Chatting

### PR DESCRIPTION
**Pull request by JaySeeJC**
Created basic framework for sending and receiving private messages. Group chats NOT implemented.

### CODE REVIEW

File: brewchat/src/main/java/com/example/brewchat/ChatService.java [hunk 1]
```diff
@@ -96,7 +101,20 @@ public void login(String username, String password) {
         QBAuth.createSession(user, new QBEntityCallbackImpl<QBSession>() {
             @Override
             public void onSuccess(QBSession session, Bundle params) {
-                EventBus.getDefault().post(new UserLoggedEvent());
+                user.setId(session.getUserId());
+                QBChatService.getInstance().login(user, new QBEntityCallbackImpl() {
```
Comment (jayseejc) [inline brewchat/src/main/java/com/example/brewchat/ChatService.java]: Was getting a nullpointer when trying to do anything with the contacts. Needed to make sure we were logged into the ChatService and not just the general QbickBlox session.

File: brewchat/src/main/java/com/example/brewchat/ChatService.java [hunk 2]
```diff
@@ -96,7 +101,20 @@ public void login(String username, String password) {
...
+                        QBChatService.getInstance().getPrivateChatManager()
+                                .addPrivateChatManagerListener(ChatService.this);
+                        EventBus.getDefault().post(new UserLoggedEvent());
+                    }
+
+                    @Override
+                    public void onError(List errors) {
+                        EventBus.getDefault().post(new AuthenticationErrorEvent(errors));
+                    }
+                });
```
Comment (JaySeeJC) [inline brewchat/src/main/java/com/example/brewchat/ChatService.java:125]: Was getting a nullpointer when trying to do anything with the contacts. Needed to make sure we were logged into the ChatService and not just the general QuickBlox session.

**Comment by jephillips** [comment]
+1

**Comment by JaySeeJC** [comment]
Future implementations will also have sendMessage(Group), so there weill be a differentiation from sendMessage(User)

**Comment by cjkennedyo** [comment]
+1

**Pull request merged**